flytekit/bin/entrypoint.py

```diff
@@ -198,47 +199,49 @@ def _handle_annotated_task(task_def: PythonTask, inputs: str, output_prefix: str
 
 
 @_scopes.system_entry_point
-def _execute_task(task_module, task_name, inputs, output_prefix, raw_output_data_prefix, test):
+def _execute_task(task_module, task_name, loader_args, inputs, output_prefix, raw_output_data_prefix, test):
     with _TemporaryConfiguration(_internal_config.CONFIGURATION_PATH.get()):
         with _utils.AutoDeletingTempDir("input_dir") as input_dir:
             # Load user code
 
             task_module = _importlib.import_module(task_module)
             task_def = getattr(task_module, task_name)
 
-            # Everything else
-            if not test and not isinstance(task_def, PythonTask):
-                local_inputs_file = input_dir.get_named_tempfile("inputs.pb")
-
-                # Handle inputs/outputs for array job.
-                if _os.environ.get("BATCH_JOB_ARRAY_INDEX_VAR_NAME"):
-                    job_index = _compute_array_job_index()
-
-                    # TODO: Perhaps remove.  This is a workaround to an issue we perceived with limited entropy in
-                    # TODO: AWS batch array jobs.
-                    _flyte_random.seed_flyte_random(
-                        "{} {} {}".format(_random.random(), _datetime.datetime.utcnow(), job_index)
+            if not test:
+                if isinstance(task_def, PythonTask):
+                    _handle_annotated_task(task_def, inputs, output_prefix, raw_output_data_prefix)
+                elif isinstance(task_def, TaskLoader):
+                    _task_def = task_def.load_task(loader_args=loader_args)
```

Comment:
So the idea is that task_module and task_name will refer to the TaskLoader instead? that's a bit odd I think... I think these should be new & different flags instead....